backend/api/handlers/application/GetTutorApplicationHandler.go

```diff
@@ -0,0 +1,69 @@
+package application
+
+import (
+	"LiteracyLink.com/backend/api/model"
+	"LiteracyLink.com/backend/auth"
+	"LiteracyLink.com/backend/db"
+	"errors"
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"golang.org/x/crypto/bcrypt"
+	"gorm.io/gorm"
+	"net/http"
+)
+
+// No idea if this is this will work tbh, but it looks like it could haha
+
+func GetTutorApplicationHandler(c *gin.Context) {
+	var request model.AppRequest
+	err := c.BindJSON(request)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"status":  "error",
+			"message": fmt.Sprintf("Error while binding tutor application request: %g", err),
+		})
+	}
+
+	dbc := c.MustGet("db").(*gorm.DB)
+	app, err := db.GetApp(request, dbc)
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			c.JSON(http.StatusBadRequest, gin.H{
+				"status":  "failed",
+				"message": fmt.Sprintf("no application exists"),
+			})
+			return
+		} else {
+			c.JSON(http.StatusBadRequest, gin.H{
+				"status":  "failed",
+				"message": fmt.Sprintf("ERROR: %g", err),
+			})
+			return
+		}
+	}
+
+	err = ((string(app.ParentId) != request.Parent) || (string(app.ChildId) != request.Child) || (string(app.DesiredSemesterId) != request.Semester))
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"status":  "failed",
+			"message": fmt.Sprintf("wrong indentifying info"),
+		})
+	}
+
+	jwt, err := auth.GenerateJWT(app.AppForTutId)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"status":  "failed",
+			"message": fmt.Sprintf("ERROR: %g", err),
+		})
+		return
+	}
+
+	c.JSON(http.StatusOK, gin.H{
+		"status":  "success",
+		"token":   jwt,
+		"message": fmt.Sprintf("TODO: make func to get application for: %s on behalf of: %s during: %s", request.Parent, request.Child, request.Semester),
```

Comment:
The message probably isn't needed here and we can remove token as well. We also need to add the application struct  to the json body so the front end can use the info

Reply:
think it would look like 
`c.JSON(http.StatusOK, gin.H{
		"status":  "success",
		app
}`